### ISSUE TITLE
Check collection dependency version strings

### ISSUE THREAD

**Issue by greg-hellings** [ansible/ansible-lint#1729]
##### Summary

Running in a collection does not detect invalid values for galaxy.yml's dependencies

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.11.5] 
  config file = /var/home/ghelling/.ansible.cfg
  configured module search path = ['/var/home/ghelling/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /var/home/ghelling/src/ansible_collections/oasis_roles/osp/.tox/ansible/lib/python3.9/site-packages/ansible
  ansible collection location = /var/home/ghelling/src:/var/home/ghelling/.ansible
  executable location = .tox/ansible/bin/ansible
  python version = 3.9.6 (default, Jul 16 2021, 00:00:00) [GCC 11.1.1 20210531 (Red Hat 11.1.1-3)]
  jinja version = 3.0.1
  libyaml = True

ansible-lint --version
ansible-lint 5.1.3 using ansible 2.11.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Fedora Silverblue 34

##### STEPS TO REPRODUCE

```yaml
# Rest of galaxy.yml here
dependencies:
  openstack.cloud: ''
```

##### Desired Behaviour
This string is invalid. At the time of attempting to import to Galaxy, the above dependency results in the following error:
`ERROR! Galaxy import process failed: Invalid collection metadata. Dependency version spec range invalid: openstack.cloud  (Code: None)`

ansible-lint should be able to check at least some, if not all, of the versions of dependency strings that are invalid. However, at present, it does not raise any issues with invalid values such as an empty string.

##### Actual Behaviour
ansible-lint passes successfully

**Comment by ssbarnea** [comment]
Depends-On: https://github.com/ansible-community/ansible-lint/issues/1803
